haskellPackages: update stackage and hackage

maralorn opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

More information about Haskell packages in nixpkgs can be found [in the nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#haskell).

---

This is the follow-up to #331380. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

maralorn committed:
haskellPackages: stackage LTS 22.31 -> LTS 22.33

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

maralorn committed:
all-cabal-hashes: 2024-07-31T18:11:52Z -> 2024-08-19T17:17:03Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

maralorn committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

github-actions[bot] committed:
Merge master into haskell-updates

ncfavier committed:
agdaPackages._1lab: unstable-2024-03-07 -> unstable-2024-08-05

Updates to a version compatible with Agda 2.7.0.

sternenseemann committed:
haskellPackages.Cabal_3_{10_3,12_1}_0: use process 1.6.22.0

sternenseemann committed:
haskellPackages.feedback: use safe-coloured-text-layout 0.2.0.1

emilazy committed:
haskellPackages.opencv{,-extra}: remove overrides

These packages have been broken since before the 24.05 release. There
is a pull request open to update them to a newer OpenCV and get them
working again: <https://github.com/LumiGuide/haskell-opencv/pull/157>.

emilazy committed:
haskellPackages.ffmpeg-light: remove override

This package has been marked as broken for a year. If anyone wants to
fix it, there is a pull request updating it to the current FFmpeg API:
<https://github.com/acowley/ffmpeg-light/pull/72>.

alexfmpe committed:
haskellPackages.entropy: build with js backend

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
Merge pull request #336209 from alexfmpe/ghc-js-entropy

haskellPackages.entropy: build with js backend

iblech committed:
agdaPackages.standard-library: 2.1 -> 2.1.1-rc2

ncfavier committed:
Merge pull request #336027 from ncfavier/1lab

agdaPackages._1lab: unstable-2024-03-07 -> unstable-2024-08-05

ncfavier committed:
Merge pull request #336351 from iblech/patch-agda-stdlib-2.1.1rc2

Unbreak Agda's standard library by bumping to 2.1.1-rc2

github-actions[bot] committed:
Merge master into haskell-updates

ncfavier committed:
agdaPackages.agda-prelude: unstable-2023-10-04 -> unstable-2024-08-22

ncfavier committed:
Merge pull request #336504 from ncfavier/agda-prelude

agdaPackages.agda-prelude: unstable-2023-10-04 -> unstable-2024-08-22

maralorn committed:
Merge pull request #336110 from emilazy/push-vlmoxqvopuvv

haskellPackages.{ffmpeg-light,opencv{,-extra}}: remove overrides

SamirTalwar committed:
haskell.ghc910: overrides for cabal-install and haskell-language-server.

github-actions[bot] committed:
Merge master into haskell-updates

SamirTalwar committed:
Override Cabal for cabal-install < 9.10.2.

maralorn committed:
Merge pull request #336630 from SamirTalwar/ghc-9.10

haskell.ghc910: overrides for cabal-install and haskell-language-server.

github-actions[bot] committed:
Merge master into haskell-updates

aidalgol committed:
oama: add to haskell-updates Hydra jobset

sternenseemann commented:
@ofborg eval

sternenseemann committed:
Merge branch master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
Merge pull request #337168 from aidalgol/oama-in-haskell-updates

oama: add to haskell-updates Hydra jobset

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.weeder: restrict to < 2.9

2.9.0 gives us a test failure and no extra compatibility, so let's wait
for that to sort itself out: https://github.com/ocharles/weeder/issues/176

sternenseemann committed:
haskellPackages.shakespeare: 2.1.1 -> 2.1.0.1

Version 2.1.1 is deprecated and has failing tests

sternenseemann committed:
git-annex: update sha256 for 10.20240808

sternenseemann committed:
haskellPackages.eventlog2html: drop upstreamed patches

github-actions[bot] committed:
Merge master into haskell-updates

eldritch-cookie committed:
haskell.packages.ghc910.haskell-language-server: fix package

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

maralorn committed:
haskellPackages.opencv-extra: Mark broken

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
haskell.packages.ghc910.haskell-language-server: fix package (#337720)

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
oama: ignore erroneous references on aarch64-darwin

The use of the Paths_ module causes erroneous references on
aarch64-darwin (due to a lack of dead code elimination), so we can't use
justStaticExecutables.

See #318013.

sternenseemann committed:
haskellPackages.{spdx,puresat}: relax lower bounds in new releases

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann commented:
`hydra-report.hs` requires `NIX_PATH=nixpkgs=.` at the moment due to #338097 until the channel has advanced.

sternenseemann committed:
haskellPackages.sequence-formats: disable tests due to missing files

sternenseemann committed:
haskellPackages.streamly-zip: provide correct package for libzip

sternenseemann committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1808565](https://hydra.nixos.org/eval/1808565) of nixpkgs commit [179b0a9](https://github.com/NixOS/nixpkgs/commits/179b0a961bbf76c374331d5952a7415656639d2f) as of 2024-08-31 14:05 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.postgresql-libpq: >= 0.10.2 and deps are broken

postgresql-libpq >= 0.10.2 invented a frankly bizarre way to discover
the libpq dependency of never actually declaring the dependency in any
cabal file, but instead discovering during configure whether it happens
to be installed anyways. Obviously this will require special
consideration from our side. (Probably we'll continue using pkg-config
which requires us to manually add pkg-config and libpq to the build
environment of postgresql-libpq-pkgconfig.)

Since Stackage LTS doesn't contain >= 0.10.2 yet, I'll ignore this for
now as I lack the patience at the moment.

sternenseemann committed:
haskellPackages.sdl2-ttf: move pkg-config override to proper place

sternenseemann committed:
haskellPackages.gi-gtk_4_0_9: ensure override is applied correctly

sternenseemann committed:
haskellPackages.sdl2: work around argv limit

sternenseemann committed:
haskellPackages.kqueue: apply patch for GHC 9.6 core libs

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
Merge branch master into haskell-updates
